draft-ietf-tls-dtls-connection-id.md

```diff
@@ -345,38 +345,32 @@ data the following modification is made to the additional data calculation.
                       length_of_DTLSInnerPlaintext;
 ~~~
 
-# Peer Address Update
+# Peer Address Update {#peer-address-update}
 
-When a CID record is received with a different source address than the
-one currently associated with the DTLS connection, the receiver SHALL
-NOT update its view of the peer's address with the source address
-specified in the enclosing UDP packet unless both conditions are met:
+When a datagram with a CID is received that has a source address different than the one currently associated with the DTLS connection, the receiver MUST NOT replace the source address currently stored in the association with the source address specified in the received datagram unless the following conditions are met:
 
-- The records in the enclosing datagram are cryptographically
-  validated;
-- The records are "newer" (in terms of their epoch and sequence
-  number) than the last record that successfully updated the peer
-  address.
+- The received datagram has been cryptographically verified using the DTLS record layer processing procedures.
+
+- The received datagram is "newer" (in terms of their epoch and sequence
+number) than the last datagram that successfully updated the source address.
+This condition ensures that replayed datagrams cannot be used to perform address 
+updates. Note that this condition does not help when the attacker is able to 
+replay datagrams that arrive earlier at the DTLS peer than the original datagram.
+
+- There is a strategy for performing a perform some reachability test. 
+No such test is defined in this	specification; {{!I-D.tschofenig-tls-dtls-rrc}} 
+defines an extension for such a DTLS protocol extension.
 
 The above ensures correctness of the protocol in presence of packet
 reordering at the network layer, while also thwarting a man-on-the-side
 attacker attempting to use spoofed or replayed records to reroute return
 traffic.
 
-However, this mechanism cannot stop an active man-in-the-middle with the
-ability to manipulate the source two-tuple to potentially turn the
-receiver into a backscatter source for a DDoS attack.  In order to
-counter this kind of attacker, an address validation protocol like the
-one described in {{!I-D.tschofenig-tls-dtls-rrc}} is needed.
-
-Since this document does not define an in-protocol peer validation
-procedure, implementations that do not already offer the mechanism
-described in {{!I-D.tschofenig-tls-dtls-rrc}} MUST, on CID-enabled
-sessions, report any peer address update to their users.  When delivered
-such an event, the user can trigger an application layer-specific
-address validation mechanism, for example one that is based on
-successful exchange of minimal amount of ping-pong traffic with the
-peer.
+Implementations that offer an application layer return routability check 
```

Comment:
turned this way sounds a bit funny.  I'd rather say that "[...] if the implementation offers *peer address update detection*, it MUST report any peer address update on CID-enabled sessions to applications"

Reply:
I merged your text with the text from Martin and replaced the current paragraph